encfs/RawFileIO.cpp

```diff
@@ -269,6 +270,14 @@ int RawFileIO::truncate(off_t size) {
     knownSize = true;
   }
 
+  if (fd >= 0 && canWrite) {
```

Comment:
Because `fdatasync` (or `fsync`) could modify `errno` set by `truncate`, and what we want in `errno` here is the return code of `truncate`, not `*sync`.